index.js

```diff
@@ -1,19 +1,69 @@
 /* globals require, __dirname, process */
 
 const path = require('path');
-const fs = require('fs');
+const { promises: fs } = require('fs');
 const builder = require("adguard-hostlists-builder");
+const SVGParser = require('svg-parser');
+
+// build services.json from services folder
+(async() => {
+    try {
+        // variable to store json's objects
+        let servicesData = []
+        // array with all service names
+        const dirNames = await fs.readdir('./services');
+        // collect data from service files
+        const collectData = dirNames.map(async(dirName) => {
+            // get file path for reading data
+            const filePath = path.resolve(`./services/${dirName}`)
+            // get data from service file
+            const fileData = await fs.readFile(filePath)
+            // parse JSON string
+            const fileDataObject = JSON.parse(fileData);
+            // collect data in array
+            servicesData.push(fileDataObject);
+        })
+
+        await Promise.all(collectData);
+
+        // stores services.json file content
+        let services = {}
+        // sort object with services
+        const sortedServicesData = servicesData.sort((a, b) => a.id.localeCompare(b.id));
+        // check if SVG is valid
+        sortedServicesData.map((serviceObject) => {
+            try {
+                const svgTree = SVGParser.parse(serviceObject.icon_svg);
+                svgTree.children[0].tagName === 'svg';
+                return true
```

Comment:
looks like linter should be added

```suggestion
                return true;
```